omod/src/main/java/org/openmrs/module/mirebalais/rest/LabLabelPrinterRestController.java

```diff
@@ -36,10 +52,33 @@ public Object printLabel(
         SimpleObject response = new SimpleObject();
 
         if ( StringUtils.isNotBlank(patientUuid) ) {
+
             Patient patient = patientService.getPatientByUuid(patientUuid);
             Location location = locationService.getLocationByUuid(sessionLocationUuid);
-            if ( patient != null && location != null){
-                response = (SimpleObject) ConversionUtil.convertToRepresentation(patient, Representation.DEFAULT);
+
+            if (patient != null && location != null){
+
+                Printer printer = printerService.getDefaultPrinter(location, PrinterType.LABEL);
+
+                if (printer != null) {
+                    Map<String, Object> paramMap = new HashMap<String, Object>();
+                    ZplLabLabelTemplate zplLabLabelTemplate = Context.getRegisteredComponent("zplLabLabelTemplate", ZplLabLabelTemplate.class);
```

Comment:
Since it was only a few lines of code, I didn't end up making another service, I took the shortcut and just generated the label within the REST controller and then printed via the print service.